herast/tree/patterns/expressions.py

```diff
@@ -303,6 +273,18 @@ def children(self):
 		return (self.first_operand, self.second_operand)
 
 
+class AbstractMemOpPat(ExpressionPat):
+	"""Abstract class for Memory operations."""
+	def __init__(self, mem_object, offset=None, **kwargs):
+		super().__init__(**kwargs)
+		self.mem_object = mem_object
+		self.offset = offset
+
+	@ExpressionPat.parent_check
+	def check(self, expression, ctx: PatternContext) -> bool:
+		return self.mem_object.check(expression.x, ctx) and (self.offset is None or self.offset == expression.m)
```

Comment:
It is preferred to check offset first and then pattern in a separate check. A bit better for perfomance.

Reply:
No problem